kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -514,4 +514,12 @@ object KyuubiConf {
     .transform(_.toUpperCase(Locale.ROOT))
     .checkValues(ShareLevel.values.map(_.toString))
     .createWithDefault(ShareLevel.USER.toString)
+
+   val CONNECTION_RELEASE_ON_CLOSE: ConfigEntry[Boolean] = buildConf("connection.release.onClose")
```

Comment:
Do we really need this conf? An unreleased engine for connection mode seems a bug to me

Reply:
Just consider to leave a legacy config to provide a fallback way.